exp/addons/internal/controllers/clusterresourceset_helpers.go

```diff
@@ -142,10 +135,29 @@ func (r *ClusterResourceSetReconciler) getOrCreateClusterResourceSetBinding(ctx
 			return nil, errors.Wrapf(err, "failed to create clusterResourceSetBinding for cluster: %s/%s", cluster.Namespace, cluster.Name)
 		}
 	}
-
 	return clusterResourceSetBinding, nil
 }
 
+// ensureOwnerRefs ensures Cluster and ClusterResourceSet owner references are set on the ClusterResourceSetBinding.
+func ensureOwnerRefs(clusterResourceSetBinding *addonsv1.ClusterResourceSetBinding, clusterResourceSet *addonsv1.ClusterResourceSet, cluster *clusterv1.Cluster) []metav1.OwnerReference {
+	ownerRefs := clusterResourceSetBinding.OwnerReferences
+	clusterResourceSetBinding.OwnerReferences = util.EnsureOwnerRef(ownerRefs, metav1.OwnerReference{
```

Comment:
should this be 
```suggestion
	ownerRefs = util.EnsureOwnerRef(ownerRefs, metav1.OwnerReference{
```
...
```suggestion
	return ownerRefs
```
In order to avoid side effects

Reply:
Should be fixed now.